### PR TITLE
Add option to allow you to control if a case-section should be intented if it contains a block.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/20009

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide 

1. will rebase against 15.6
2. working on tests now.

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by tannergooding** [comment]
Mentioned it in slack, but `IndentSwitchCaseSectionWhenBlock` needs to be `true` by default in order to preserve existing behavior as the default.

**Comment by CyrusNajmabadi** [comment]
> IndentSwitchCaseSectionWhenBlock needs to be true by default

Have set to 'true'.

**Comment by alrz** [comment]
FWIW, alternatively we could just say "indent switch section blocks" -- wherever they appear, giving
```cs
case 0:
{
    break;
}
{
    break;
}
    break;
case 1:
    break;
```
With current behavior, someone who actually wants to indent case section body, would end up with this:
```cs
case 0:
{
}
break;
case 1:
    break;
```
That does not seem to reflect one preference to indent case section bodies, by virtue of using a block in the beginning.

File: src/VisualStudio/CSharp/Impl/Options/Formatting/IndentationViewModel.cs [hunk 1]
```diff
@@ -46,6 +46,21 @@ class MyClass
     }
 }";
 
+        private const string SwitchCaseWhenBlockPreview = @"
+class MyClass
+{
+    int Method(int foo){
+//[
+        switch (foo){
```
Comment (Pilchie) [inline src/VisualStudio/CSharp/Impl/Options/Formatting/IndentationViewModel.cs]: Please avoid adding more occurrences of `foo`.

File: src/Workspaces/CSharp/Portable/Formatting/CSharpFormattingOptions.cs [hunk 2]
```diff
@@ -146,6 +146,13 @@ public static partial class CSharpFormattingOptions
                 EditorConfigStorageLocation.ForBoolOption("csharp_indent_case_contents"),
                 new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.IndentSwitchCaseSection")});
 
+        public static Option<bool> IndentSwitchCaseSectionWhenBlock { get; } = new Option<bool>(
+            nameof(CSharpFormattingOptions),
+            nameof(IndentSwitchCaseSectionWhenBlock), defaultValue: true,
+            storageLocations: new OptionStorageLocation[] {
+                EditorConfigStorageLocation.ForBoolOption("csharp_indent_case_contents_when_block"),
```
Comment (jasonmalinowski) [inline src/Workspaces/CSharp/Portable/Formatting/CSharpFormattingOptions.cs]: Do we also need to update the AutomationObject for this one, for .vssettings file support?

Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/Formatting/CSharpFormattingOptions.cs]: Probably.  Thanks for reminding me.

**Comment by CyrusNajmabadi** [comment]
restest windows_release_vs-integration_prtest please

**Comment by CyrusNajmabadi** [comment]
(IO error)

**Comment by CyrusNajmabadi** [comment]
retest windows_release_vs-integration_prtest  please

**Comment by tannergooding** [comment]
Could we add a bug (or possibly just add it to this PR) tracking an EditorConfig option for this?

**Comment by Pilchie** [comment]
There's already an editorconfig option in this PR.

**Comment by tannergooding** [comment]
I'm blind and completely missed it: `csharp_indent_case_contents_when_block`

**Comment by tannergooding** [comment]
This also fixes https://github.com/dotnet/roslyn/issues/12304

**Comment by tannergooding** [comment]
This also makes https://github.com/dotnet/roslyn/pull/13264 unnecessary (it should probably be closed once this is merged)

**Pull request merged**